codelabs/getting_started/getting_started.md

```diff
@@ -0,0 +1,422 @@
+- Blockly core: The main Blockly library, which defines the basic Blockly UI and logic.
+- Built-in block definitions: Common blocks such as loops, logic, math, and string manipulation.
+- The JavaScript generator: Converts blocks into JavaScript, and contains block generators for all built-in blocks.
+- English language files: String tables for all messages on built-in blocks and the Blockly UI, in English.
+
+### Alternate imports
+
+There are many ways to import a library in JavaScript, and this tutorial does not cover all of them. For samples that show how to integrate Blockly in your project, look at the `examples` folder in [blockly-samples](https://github.com/google/blockly-samples).
+
+You can also define your imports more carefully to get [different generators](https://www.npmjs.com/package/blockly#blockly-generators) and [locales](https://www.npmjs.com/package/blockly#blockly-languages).
+
+## Create a Blockly workspace
+
+In this section you will learn how to add a workspace to your app, including how to define a toolbox.
+
+### The Blockly toolbox
+
+A Blockly workspace has two main components:
+- The area where the user assembles their blocks (the white area).
+- A toolbox that contains all blocks that are available to the user (the grey area).
+
+![image](d4_three_times.png)
+
+The toolbox may be organized into categories, and may contain both single blocks and groups of blocks. A well-organized toolbox helps the user to explore the available blocks and understand the capabilities of the underlying system.
+
+A toolbox is defined in XML and passed into the workspace constructor through an options struct.
+
+For more information on this XML format and toolbox configuration, including category creation, click <a href="https://developers.google.com/blockly/guides/configure/web/toolbox">here</a>.
+
+### Define the toolbox
+
+Add the structure of the toolbox just after the `blockly-div`:
+
+```xml
+<xml id="toolbox" style="display: none">
+  <block type="controls_repeat_ext">
+    <value name="TIMES">
+      <shadow type="math_number">
+        <field name="NUM">5</field>
+      </shadow>
+    </value>
+  </block>
+</xml>
+```
+
+This XML defines a toolbox with a single "repeat loop" block.
+
+We set the display style of the toolbox to `none`, because we do not intend to display the XML structure on our web page - it will be just used to construct the toolbox programmatically.
+
+### Injection
+
+Adding a Blockly workspace to a page is called *injection*, because the workspace is injected into a `div` that lready exists on the page.
+
+The function to call is `Blockly.inject(container, options)`, which takes two arguments:
+- `container` is where the Blockly workspace should be placed on the page. It can be an `Element`, an ID string, or a CSS selector.
+- `options` is a dictionary of configuration options.
+
+For this codelab we will inject into a div with the id `"blockly-div"`, which you can find in `index.html`:
```

Comment:
In the interest of keeping things standardized, all the `demos/` and blockly-samples examples / plugins, use `blocklyDiv` as the blockly id. Might be good to keep standardize on that.

Reply:
Done.